jwst/ramp_fitting/ols_fit.py

```diff
@@ -0,0 +1,3491 @@
+        The output final rate of each pixel
+
+    new_model.dq : 2-D DQflag
+        The output pixel dq for each pixel
+
+    new_model.var_poisson : 2-D float32
+        The variance in each pixel due to Poisson noise
+
+    new_model.var_rnoise : 2-D float32
+        The variance in each pixel due to read noise
+
+    new_model.err : 2-D float32
+        The output total variance for each pixel
+
+    int_data : 3-D float32
+        The rate for each pixel in each integration
+
+    int_dq : 3-D float32
+        The pixel dq flag for each integration
+
+    int_var_poisson : 3-D float32
+        The variance of the rate for each integration due to Poisson noise
+
+    int_var_rnoise : 3-D float32
+        The variance of the rate for each integration due to read noise
+
+    int_err : 3-D float32
+        The total variance of the rate for each integration
+
+    int_int_times : 3-D
+        The total time for each integration
+
+    opt_slope : 4-D float32
+        The rate of each segment in each integration
+
+    opt_sigslope : 4-D float32
+        The total variance of the rate for each pixel in each segment of each integration
+
+    opt_var_poisson : 4-D float32
+        The Poisson variance of the rate for each pixel in each segment of each integration
+
+    opt_var_rnoise : 4-D float32
+        The read noise variance of the rate for each pixel in each segment of each integration
+
+    opt_yint : 4-D float32
+        The y-intercept for each pixel in each segment of each integration
+
+    opt_sigyint : 4-D float32
+        The variance for each pixel in each segment of each integration
+
+    opt_pedestal : 4-D float32
+        The zero point for each pixel in each segment of each integration
+
+    opt_weights : 4-D float32
+        The weight of each pixel to use in combining the segments
+
+    opt_crmag : 4-D float32
+        The magnitude of each CR in each integration
+
+    actual_segments : int
+        The actual maximum number of segments in any integration
+
+    actual_CRs : int
+        The actual maximum number of CRs in any integration
+    """
+    # Package image data into a RampModel
+    input_model = RampModel()
+
+    input_model.data = data
+    input_model.err = err
+    input_model.groupdq = groupdq
+    input_model.pixeldq = inpixeldq
+
+    # Capture exposure and instrument data into the RampModel
+    input_model.meta.instrument.name = instrume
+
+    input_model.meta.exposure.frame_time = frame_time
+    input_model.meta.exposure.ngroups = ngroups
+    input_model.meta.exposure.group_time = group_time
+    input_model.meta.exposure.groupgap = groupgap
+    input_model.meta.exposure.nframes = nframes
+    input_model.meta.exposure.drop_frames1 = dropframes1
+
+    # Compute ramp fitting
+    new_model, int_model, opt_res = ols_ramp_fit_single(
+        input_model, int_times, buffsize, save_opt, readnoise_2d, gain_2d, weighting)
+
+    if new_model is None:
+        return None * 22
```

Comment:
It does cause a failure.  I will change this, but this brings up a larger issue of what to do if a slice fails.  Starting at line 

https://github.com/spacetelescope/jwst/blob/7908fbe3996fe343675ecc9fac2e90dd431dcb9e/jwst/ramp_fitting/ols_fit.py#L173 

the processing assumes non-`None` values and continues that way for the loop to populate the output models.  If any particular slice returns `None` values, should the final results be None or on that slice?  If the latter, the output models need to be created using a different method because it's possible for the first slice to be `None`.

These issues will be corrected in JP-2037, since this is the multiprocessing section of the code.

Reply:
I would think that if any slice fails for some reason, the whole process should fail (i.e. don't create an output that has gaps in it).